sql/src/test/java/io/crate/integrationtests/SysSegmentTableInfoTest.java

```diff
@@ -0,0 +1,92 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.core.Is.is;
+
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1)
+public class SysSegmentTableInfoTest extends SQLTransportIntegrationTest {
+
+
+    @After
+    public void deleteTestData() {
+        execute("drop table t1;");
+    }
+
+    @Test
+    public void test_retrieve_segment_information_for_1_shard_0_replica_1_segment() {
+        execute("create table t1 (id INTEGER, name STRING) clustered into 1 shards with (number_of_replicas = 0)");
+        ensureYellow();
+        execute("insert into t1 values (1, 'test')");
+        refresh();
+        execute("select * from sys.segments");
+        assertThat(response.rowCount(),is(1L));
+        assertThat(response.rows()[0].length,is(15));
+
+        execute("select table_name, shard_id, segment_name, generation, num_docs, deleted_docs, size, memory, " +
+                "committed, search, compound, version, _node, attributes from sys.segments");
+
+        assertThat(response.rowCount(),is(1L));
+        assertThat(response.rows()[0][0],is("t1"));
+        assertThat(response.rows()[0][1],is(0));
+        assertThat(response.rows()[0][2],is("_0"));
+        assertThat(response.rows()[0][3],is(0L));
+        assertThat(response.rows()[0][4],is(1));
+        assertThat(response.rows()[0][5],is(0));
+        assertThat((Long) response.rows()[0][6], greaterThan(3000L));
+        assertThat((Long) response.rows()[0][7], greaterThan(900L));
+        assertThat(response.rows()[0][8],is(false));
+        assertThat(response.rows()[0][9],is(true));
+        assertThat(response.rows()[0][10],is(true));
+        assertThat(response.rows()[0][11],is(Version.CURRENT.luceneVersion.toString()));
+        Map node = (Map) response.rows()[0][12];
+        assertNotNull(node);
+        assertThat(node.size(), is(2));
+        assertTrue(node.containsKey("name"));
+        assertTrue(node.containsKey("id"));
+        Map attributes = (Map) response.rows()[0][13];
+        assertNotNull(attributes);
+        assertThat(attributes.size(), is(1));
+    }
+
+    @Test
+    public void test_retrieve_segment_information_for_1_shard_1_replica_1_segment() {
+        execute("create table t1 (id INTEGER, name STRING) clustered into 1 shards with (number_of_replicas = 1)");
```

Comment:
I think on test which is testing at least 2 shards (could be also 1 primary and 1 replicas) is enough as the code path to test should be the same.